feat: Charts package continued

leofarias-cpu opened the pull request:
## Summary
The `@vtex/shoreline-charts` library wraps `echarts` in a react component with consistent theming and features. 
This is a **work-in-progress** continuation of [Natan's work](https://github.com/vtex/shoreline/pull/1648) by VTEX lab@UFCG, with new features, types of charts, documentation, testing and fixes.

### Default styles for line and bar charts
Mostly improvements and fixes from the original PR. In the future more chart types will be supported, such as area charts.

### Composite charts 
Previously it was possible to make i.e. a bar chart with a line, but only one of the two would be properly themed. With the new `<ChartCompositor>` component all possible combinations of chart types and variant can be used together, all with shoreline-charts styling.

### Custom legend selection behavior
[Screencast from 2025-04-16 11-47-52.webm](https://github.com/user-attachments/assets/26227c8e-4626-427d-b80a-a7f8bb56d30a)


### Pre-processing 
Implemented with the 'seriesHooks' prop, allows for more fine grained data control at a series level. Users can pass functions and some chart types have default ones, which allows us to, for instance, make negative bars have proper rounded corners, which Echarts by itself does not do.


## Examples
### Composite chart with bars and lines
```ts
const data1: number[] = []
const data2: number[] = []
const data3: number[] = []

for (let i = 0; i < 25; i++) {
  data1.push((Math.sin(i / 5) * (i / 5 - 10) + i / 6) * 5)
  data2.push((Math.cos(i / 5) * (i / 5 - 10) + i / 6) * 5)
  data3.push((data1[i] + data2[i]) / 2)
}

return (
  <ChartCompositor
    charts={[
      {
        series: { data: data1, name: 'Bar 1' },
        chartConfig: { type: 'bar' },
      },
      {
        series: { data: data2, name: 'Bar 2' },
        chartConfig: { type: 'bar' },
      },
      {
        series: { data: data3, name: 'Average' },
        chartConfig: { type: 'line' },
      },
    ]}
    tooltip={{ type: 'line' }}
    zoom
  />
)
```
![image](https://github.com/user-attachments/assets/79588a79-bdc8-403e-a33b-6801c3441728)


### Line chart
``` ts
<Chart 
  option={{
    xAxis: {
      data: ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun']
    },
    series: {
		data: [820, 932, 901, 934, 1290, 1330, 1320],
    }
  }} 
  chartConfig={type: "bar" 
/>
```
![image](https://github.com/user-attachments/assets/a1897553-e65d-4327-9ce7-c1c3b683f23c)

beatrizmilhomem reviewed (changes requested):
Hello!

**Design QA** – Bia Oliveira documented what needs to be adjusted [here](https://www.figma.com/design/exdkWev8U39Fz6vagTvUv8/Shoreline---Data-Visualization?node-id=5114-47361).

**Information Architecture** – There are some necessary adjustments to be made in the architecture of the documentation, they are [here](https://www.figma.com/design/NGb7dSIhhfW0i0RqOztzEo/Shoreline-Documentation?node-id=2260-17574). Please reach out if you have any questions.

JRamonAlves committed:
docs(charts): remake the area chart stories

JRamonAlves committed:
docs(charts): create the docs for the area charts making

JRamonAlves committed:
fix(charts): separate area and line stories, and fix hook from area chart

leofarias-cpu committed:
feat(chart): adds green color

leofarias-cpu committed:
feat(chart): area chart and stacked variant implemented

leofarias-cpu committed:
fix: removed zoom from area chart by default

leofarias-cpu committed:
fix: compositor typing and chart title typing

leofarias-cpu committed:
docs: removed area example

JRamonAlves committed:
fix(charts): make stories using the args and not using render

JRamonAlves committed:
fix(charts): fix zoom bar in line chart

leofarias-cpu committed:
style(chart): added names to area storybook

leofarias-cpu committed:
style(chart): added names to area storybook

JRamonAlves committed:
fix(charts): make zoom bar default

JRamonAlves committed:
fix(charts): fix bar gap hook stories

JRamonAlves committed:
refactor(charts): creates a way to force the chart reloading, making zoom bar disapear

JRamonAlves committed:
refactor(charts): creates a way to force the chart reloading, making zoom bar disapear

leofarias-cpu committed:
style(chart): renamed some bar storybooks

leofarias-cpu committed:
style(chart): renamed default area variant to overlapping

leofarias-cpu committed:
fix(chart): removed zoom from default in area chart

leofarias-cpu committed:
docs(chart): added temporary chart examples

leofarias-cpu committed:
Merge branch 'vtex:main' into main

leofarias-cpu committed:
Merge branch 'feature/charts-package' of https://github.com/leofarias-cpu/shoreline-charts

leofarias-cpu committed:
docs(charts): restructured data visualization~

leofarias-cpu committed:
docs(charts): restructured data visualization

leofarias-cpu committed:
docs(chart): fix typo

leofarias-cpu committed:
docs(chart): renames bar page

leofarias-cpu committed:
docs(chart): re-added design bar chart docs and fixed an example

leofarias-cpu committed:
docs(chart): re-added design bar chart docs and fixed an example

JRamonAlves committed:
refactor(charts): remake of tooltip for the area charts based on designer return

JRamonAlves committed:
refactor(charts): remake of tooltip for the area charts based on designer return

JRamonAlves committed:
refactor(charts): rename for reverse in the tooltip options

JRamonAlves committed:
refactor(charts): rename for reverse in the tooltip options

leofarias-cpu committed:
docs(chart): fixes some examples

leofarias-cpu committed:
docs(chart): fixes some examples

leofarias-cpu committed:
feat(chart): start of legend reimplementation

leofarias-cpu committed:
feat(chart): area chart colors updated and checkboxLegend fully functional

JRamonAlves committed:
fix(charts): redo the check box legend behavior to make it work with synced charts

JRamonAlves committed:
fix(charts): redo the check box legend behavior to make it work with synced charts

leofarias-cpu committed:
fix(chart): checkbox legend correctly works for every chart again

JRamonAlves committed:
feat(chart): creates function to activate or not zoom function

JRamonAlves committed:
feat(chart): creates function to activate or not zoom function

JRamonAlves committed:
fix(chart): makes trigger on axis the default

JRamonAlves committed:
fix(chart): makes trigger on axis the default

leofarias-cpu committed:
fix(chart): no longer flickers and re renders on window size change

leofarias-cpu committed:
refactor(chart): added braces to conditionals

leofarias-cpu committed:
refactor(chart): added braces to conditionals

leofarias-cpu committed:
feat(chart): added optional checkbox visual

leofarias-cpu committed:
feat(chart): added optional checkbox visual

leofarias-cpu committed:
fix(chart): only line chart has zoom by default

leofarias-cpu committed:
refactor(chart): setAreaColors is now cleaner and the gradient is identical to figma

leofarias-cpu committed:
fix(chart): changed area chart area color back

leofarias-cpu committed:
fix(chart): changed area chart area color back

JRamonAlves committed:
fix(charts): fix the sync legend problem with the same name

leofarias-cpu committed:
fix(chart): charts are synced when rendered, not later

leofarias-cpu committed:
style(chart): rename onFinished to onRendered

leofarias-cpu committed:
style(area-chart): rewrote xaxis in english

leofarias-cpu committed:
fix(chart-line): remove zoom from being enabled by defautl

leofarias-cpu committed:
test(chart-area): swap area story for the equivalent area example that is cleaner

leofarias-cpu committed:
fix(chart-area): correct colors and opacity of area

leofarias-cpu committed:
fix(chart): remove padding

leofarias-cpu committed:
fix(chart-area): remove gap between yaxis and areas

JRamonAlves committed:
docs(chart-line): fix name to trend line chart

JRamonAlves committed:
docs(chart-line): make the chart has one more line in gray-6

JRamonAlves committed:
refactor(chart-compositor): make chart compositor takes more functions from the original chart

leofarias-cpu committed:
feat(chart-line): speed up animation

leofarias-cpu committed:
fix(chart): add 8 right padding to prevent cutoff of chart elements

leofarias-cpu committed:
docs(chart): reorganize chart and remove non design best pratices

JRamonAlves committed:
docs(chart): adds loading state examples for charts

JRamonAlves committed:
Merge branch 'feature/charts-package' of https://github.com/theo-rodrigues/shoreline-charts into feature/charts-package

JRamonAlves committed:
fix(chart-line): typo mistakes and makes the trend line gray

leofarias-cpu committed:
fix(chart): turning off checkboxlegend behaviour works again

leofarias-cpu committed:
docs(chart): reorganize chart and remove non design best pratices

leofarias-cpu committed:
feat(chart): make titles centered

leofarias-cpu committed:
test(chart): add animation storybook

leofarias-cpu committed:
docs(chart): update changing styling and add a warning about not changing them

JRamonAlves committed:
fix(chart): fix names and storybooks names

JRamonAlves committed:
Merge branch 'feature/charts-package' of https://github.com/theo-rodrigues/shoreline-charts into feature/charts-package

leofarias-cpu committed:
chore(package.json): move echarts from a devDepencie to a regular one

JRamonAlves committed:
test(chart): add story to test animations

JRamonAlves committed:
fix(chart): remove test from the story

JRamonAlves committed:
fix(charts): fix names and where should be the story of animation test

leofarias-cpu committed:
test(chart): refactor and rename animation easing functions example

leofarias-cpu committed:
test(chart): tooltip story now renders legend again

leofarias-cpu committed:
feat(chart): add new tooltip and change tooltip story to match new figma layout

leofarias-cpu committed:
fix(chart): fix imports in chart component

leofarias-cpu committed:
docs(chart): move loading state examples further down

leofarias-cpu committed:
docs(chart): rename bar best practices

leofarias-cpu committed:
docs(chart): fix duplicate compositor page and remove optional props section

leofarias-cpu committed:
fix(chart): change to use shoreline color when possible

leofarias-cpu committed:
feat(chart): checkbox legend visuals uses series colors directly

leofarias-cpu committed:
fix(chart-area): fix area colors

leofarias-cpu committed:
fix(chart-area): fix area line colors

leofarias-cpu committed:
fix(chart-compositor): fix checkbox legend colors

leofarias-cpu committed:
refactor(chart): move legend color attribute get closer to where its used

leofarias-cpu committed:
test(chart): remove debug square

leofarias-cpu committed:
fix(chart): add tooltip min and max height, use tokens for spacing

leofarias-cpu committed:
fix(chart): fix spacing between tooltip colored line and text

leofarias-cpu committed:
fix(chart): confine tooltip to chart boundaries

leofarias-cpu committed:
refactor(chart): simplify conditionals in checkbox legend

JRamonAlves committed:
feat(charts): calls the onEvents functions in addition to the intern event handlers

leofarias-cpu committed:
style(chart): clarifiy conditional around legendvisuals

leofarias-cpu committed:
test(chart-line): remove unecessary type setting for series

leofarias-cpu committed:
style(chart): rename eventsAdaption to eventsAdapter

leofarias-cpu committed:
feat(chart-bar): add early implementation of stacked bar chart

leofarias-cpu committed:
fix(chart): removed uncessary cloning in chart hooks

leofarias-cpu committed:
fix(chart): fix optionHooks jsdocs

leofarias-cpu committed:
docs(charts): restyled some examples for more consistency

leofarias-cpu committed:
docs(chart): fix best practices title and rename bar variant from default to vertical

JRamonAlves committed:
feat(chart-legend): creates the first version of the legend of the chart

JRamonAlves committed:
fix(chart-zoom): makes compositor use correct zoom functions and turn off tests with zoom

leofarias-cpu committed:
feat(funnel-chart): add funnel chart prototype

JRamonAlves committed:
feat(chart-legend): add states to the legend and their treatment

leofarias-cpu committed:
fix(chart-area): now allows changing colors

leofarias-cpu committed:
fix(chart-area): remove console log

JRamonAlves committed:
feat(chart-legend): makes legend behaviour and states

JRamonAlves committed:
refactor(chart-legend): refactor code for more clean usage and logics on the behaviour

JRamonAlves committed:
refactor(chart-legend): fix sync test and delete not used functions

JRamonAlves committed:
fix(chart-legend): fix the size of the legend due to designer reviews

leofarias-cpu committed:
feat(chart-area): change area color to match line color

JRamonAlves committed:
fix(chart-legend): fix the legend font and check icon size

JRamonAlves committed:
fix(chart-legend): makes the legend button not focus during the mouse down

JRamonAlves committed:
fix(chart-legend): fix according to the Pull Request Review

leofarias-cpu committed:
feat(chart-bar-stacked): refine implementation

leofarias-cpu committed:
fix(chart-bar-stacked): correct gap size

leofarias-cpu committed:
fix(chart): using shoreline color tokens no longer flickers

leofarias-cpu committed:
feat(chart): add stacked percentage bar

JRamonAlves committed:
fix(chart-legend): fix the legend resize in small displays

leofarias-cpu committed:
fix: fixed depencies in chart useMemos and useCallback, added warn to biome

leofarias-cpu committed:
fix(chart): add css to change cursor on chart series and set stack bar gaps

leofarias-cpu committed:
fix(chart): css atribute now checks for default variant by type

leofarias-cpu committed:
feat(chart-bar-stacked percentage): y axis label now in percentage % form

JRamonAlves committed:
test(chart): make a loading example for the charts

leofarias-cpu committed:
feat(chart): user hooks now execute after default ones

JRamonAlves committed:
fix(chart-legend): add prop to whether use the checkbox legend

JRamonAlves committed:
fix(chart-legend): add hover for turned off serie

JRamonAlves committed:
Merge branch 'feature/charts-package' of https://github.com/theo-rodrigues/shoreline-charts into feature/charts-package-legend

JRamonAlves committed:
fix(chart-legend): fix merge problems

leofarias-cpu committed:
style: checkboxLegends made singular

leofarias-cpu committed:
Merge pull request #2 from theo-rodrigues/feature/charts-package-legend

feat: chart legend remake

leofarias-cpu committed:
feat(chart): initial time axis implementation

leofarias-cpu committed:
fix(chart): add locale to hooks dependencies

leofarias-cpu committed:
feat(chart): time axis format now receives format options

JRamonAlves committed:
style(chart-legend): fix border-radius

leofarias-cpu committed:
feat: init localization

leofarias-cpu committed:
feat: add localization for a lot of languages, including portuguese

JRamonAlves committed:
docs(chart-hooks): creates the first version of the hook page

JRamonAlves committed:
feat(chart-bar-stacked): always rounds the top bar, using the legend handle

theo-rodrigues committed:
fix(legend): adjust hover and overlap color for area chart

theo-rodrigues committed:
chore: remove unused variables

leofarias-cpu committed:
fix(chart-compositor): legend now works again

leofarias-cpu committed:
style: change funnel story series name

JRamonAlves committed:
fix(chart): uncomment resizing code

JRamonAlves committed:
fix(chart-legend): fix legend state conversion

JRamonAlves committed:
refactor(chart-legend): refactor legend states to use strings in order to boolean

theo-rodrigues committed:
fix(benchmark): fix components used in tests

theo-rodrigues committed:
refactor(legend): make legend handle diferent changes

leofarias-cpu committed:
docs: add page about charts synchronization

JRamonAlves committed:
feat(chart-colors): allow to the user pass a link to a color shade using a function

leofarias-cpu committed:
fix: typo in addHoverToColors

JRamonAlves committed:
fix(chart-colors): fix function docs and name

leofarias-cpu committed:
docs(chart): update changing styling to feature changing legend hover colors and checkboxLegend prop

JRamonAlves committed:
feat(chart-scatter): creates a prototype of scatter chart without design touch

JRamonAlves committed:
feat(chart-scatter): creates a prototype of scatter chart without design touch

JRamonAlves committed:
fix(chart-scatter): fix to make chart stories not random

JRamonAlves committed:
fix(chart-scatter): fix to make chart stories not random

leofarias-cpu committed:
feat: add sunburst chart and make emphasis easier to use by the user

leofarias-cpu committed:
feat(chart-sunburst): add hook to color root node

leofarias-cpu committed:
fix(chart-compositor): pass option correctly to chart

leofarias-cpu committed:
fix(chart-compositor): grid can now be modified

leofarias-cpu committed:
feat(chart-compositor): add story with sunburst

theo-rodrigues committed:
feat(chart): add example for donut chart

JRamonAlves committed:
Merge branch 'feature/chart-scatter-prototype' of https://github.com/theo-rodrigues/shoreline-charts into feature/chart-scatter-prototype

JRamonAlves committed:
fix(chart-types): fix merge problems

JRamonAlves committed:
fix(chart-compositor): fix build problems with compositor

JRamonAlves committed:
Merge pull request #4 from theo-rodrigues/feature/chart-scatter-prototype

feat: adds a scatter chart prototype

JRamonAlves committed:
test(chart-scatter): creates benchmark for scatter chart

leofarias-cpu committed:
docs(chart-area): add description for chart area

leofarias-cpu committed:
fix(chart): change optionHooks docs 'before default styling -> after default styling'

leofarias-cpu committed:
docs: removed changing styling page and changed hooks example to be closer to a real use case

leofarias-cpu committed:
feat(chart): loading state nowshows axis lines and no longer breaks the tooltip

leofarias-cpu committed:
feat(chart): turns off auto merge, making the chart less statefull

leofarias-cpu committed:
feat(chart): loading examples improved to allow manual toggling of load state

JRamonAlves committed:
feat(chart-tooltip): change the way tooltip is created

JRamonAlves committed:
fix(chart-tooltip): fix type errors

JRamonAlves committed:
fix(chart): fix chart API error

leofarias-cpu committed:
feat(chart): add chart skeletons

leofarias-cpu committed:
feat(chart-skeleton): added configuration for number of lines and columns

leofarias-cpu committed:
feat(chart-skeleton): added space for where the legend would be

leofarias-cpu committed:
docs(chart): changed loading examples to use new loadingOptions

JRamonAlves committed:
fix(chart-tooltip): remove all the lastests tooltips changes

JRamonAlves committed:
fix(chart-tooltip): remove unsed props

leofarias-cpu committed:
docs(chart-area): fixed loading state example to actually be an area chart

JRamonAlves committed:
feat(chart-scatter): make new variant for sactter. New variant shows value after position

JRamonAlves committed:
Merge branch 'feature/charts-package' of https://github.com/theo-rodrigues/shoreline-charts into feature/charts-package

JRamonAlves committed:
test(chart-scatter): test in storybook for the new variant

ThaynanAndrey commented on .github/workflows/chromatic.yml:
```diff
@@ -0,0 +1,22 @@
+name: "Chromatic"
```
Do you need this for shoreline package?

ThaynanAndrey commented on biome.json:
```diff
@@ -52,7 +52,7 @@
       },
       "correctness": {
-        "useExhaustiveDependencies": "off",
+        "useExhaustiveDependencies": "warn",
```
Is it recommended?

leofarias-cpu replied:
Yes, it is the recommended default, someone intentionally removed it but there's no discussion about this that i can find. I actually [wrote an issue](https://github.com/vtex/shoreline/issues/2111) about this earlier. [Eslint has it on by default as well](https://react.dev/reference/eslint-plugin-react-hooks).

ThaynanAndrey commented on packages/charts/README.md:
```diff
@@ -0,0 +1,7 @@
+# Shoreline Charts
+
+`shoreline-components` and `echarts` are peer dependencies of `shoreline-charts`
```
Could you add a description about the charts, beside how rot run, test and keep it?

ThaynanAndrey commented on packages/charts/src/components/chart-compositor/chart-compositor.tsx:
```diff
@@ -0,0 +1,191 @@
+ * Functions that are always called for a certain chart config
+ */
+const defaultHooks: DefaultHooks = {
+  bar: {
```
These types should be more external package to reuse instead of be here?

ThaynanAndrey commented on packages/charts/src/components/chart-skeleton/chart-skeleton.tsx:
```diff
@@ -0,0 +1,88 @@
+          ))}
+          {numLines <= 0 ? null : <ChartSkeletonLine isXAxisLine />}
+        </Flex>
+        <Flex justify="space-between" style={{ paddingLeft: '40px' }}>
```
Same issue about style. BTW, why did you put exactly 40px?

ThaynanAndrey commented on packages/charts/src/components/chart-skeleton/chart-skeleton.tsx:
```diff
@@ -0,0 +1,88 @@
+    <Flex
+      align="center"
+      gap={0}
+      style={{
```
Same about styles instead of class.

ThaynanAndrey commented on packages/charts/src/components/chart/chart.tsx:
```diff
@@ -0,0 +1,417 @@
+      }
+      hooks.push(...defaultHooks[type][checkedVariant])
+      hooks.push(...optionHooks)
+      return hooks
```
Take care with the readability. Add break lines between scopes like, between ifs and returns.

ThaynanAndrey commented on packages/charts/src/components/chart/chart.tsx:
```diff
@@ -0,0 +1,417 @@
+        if (action.type === 'toggle' && action.index < series.length) {
+          toggleSerieLegend(chart, String(series[action.index].name))
+          if (change)
+            chart.setOption(handleChanges(change, finalOptions, action))
```
Readability
```suggestion
          if (change) {
            chart.setOption(handleChanges(change, finalOptions, action))
         }
```

ThaynanAndrey commented on packages/charts/src/components/chart/chart.tsx:
```diff
@@ -0,0 +1,417 @@
+          series.forEach((s, index) => {
+            if (index === action.index) turnOnSerieLegend(chart, String(s.name))
+            else turnOffSerieLegend(chart, String(s.name))
+          })
```
```suggestion
          })
       
```

ThaynanAndrey commented on packages/charts/src/components/chart/chart.tsx:
```diff
@@ -0,0 +1,417 @@
+
+    const connectGroups = useCallback(
+      (_params?: any) => {
+        if (!group || !chartRef.current) return
```
```suggestion
        if (!group || !chartRef.current) return
        
```

ThaynanAndrey commented on packages/charts/src/components/chart/chart.tsx:
```diff
@@ -0,0 +1,417 @@
+          style={{ minWidth: 300, minHeight: 200, ...style }}
+          opts={{ renderer: renderer, locale: locale }}
+          showLoading={loading}
+          // onChartReady={(instance) => instance.resize()}
```
Ops

ThaynanAndrey commented on packages/charts/src/components/chart/locales.ts:
```diff
@@ -0,0 +1,55 @@
+import uk from 'echarts/lib/i18n/langUK'
+import vi from 'echarts/lib/i18n/langVI'
+import zh from 'echarts/lib/i18n/langZH'
+// import nbNO from 'echarts/lib/i18n/langnb-NO';
```
?

ThaynanAndrey commented on packages/charts/src/components/chart/locales.ts:
```diff
@@ -0,0 +1,55 @@
+echarts.registerLocale('AR', ar)
+echarts.registerLocale('CS', cs)
+echarts.registerLocale('DE', de)
+// echarts.registerLocale('EL', el);
```
?

ThaynanAndrey commented on packages/charts/src/components/chart/locales.ts:
```diff
@@ -0,0 +1,55 @@
+echarts.registerLocale('UK', uk)
+echarts.registerLocale('VI', vi)
+echarts.registerLocale('ZH', zh)
+// echarts.registerLocale('nb-NO', nbNO);
```
?

ThaynanAndrey commented on packages/charts/src/components/legend/legend.tsx:
```diff
@@ -0,0 +1,203 @@
+import type { ChartConfig } from '../../types/chart'
+
+export const Legend = forwardRef<LegendHandle, LegendProps>(
+  function Legend(props, ref) {
```
It's little bit confuse te name of Legend is used twice. Could you change to arrow function instead of?

ThaynanAndrey commented on packages/charts/src/components/legend/legend.tsx:
```diff
@@ -0,0 +1,203 @@
+    const { series, chartRef, ...otherProps } = props
+
+    if (!series) return
+    if (!isArray(series)) return
```
```suggestion
    if (!series || !isArray(series)) return
```

ThaynanAndrey commented on packages/charts/src/components/legend/legend.tsx:
```diff
@@ -0,0 +1,203 @@
+          return defaultColorPreset[index++]
+        }
+        if (isString(serie.color)) return serie.color
+        return serie.color[0]
```
Beware magic number

ThaynanAndrey commented on packages/charts/src/components/legend/legend.tsx:
```diff
@@ -0,0 +1,203 @@
+        })
+      },
+      [seriesState, chartRef, changeState]
+    )
```
Some tips: use meaningful names instead of "on",  "off",  etc.; avoid magic numbers; create constants like enum instead of hardcoded strings like `exclusive`, `selectAll`, etc.

ThaynanAndrey commented on packages/charts/src/components/legend/legend.tsx:
```diff
@@ -0,0 +1,203 @@
+    )
+
+    return (
+      <div data-sl-chart-legend {...otherProps} onClick={undefined}>
```
Why not Box?

ThaynanAndrey commented on packages/charts/src/components/legend/legend.tsx:
```diff
@@ -0,0 +1,203 @@
+  }
+)
+
+export function getChanges(chart: ChartConfig | null): string {
```
Could you pls document this?

ThaynanAndrey commented on packages/charts/src/components/legend/legend.tsx:
```diff
@@ -0,0 +1,203 @@
+  return change
+}
+
+export function handleChanges(
```
Documentatiom

ThaynanAndrey commented on packages/charts/src/components/legend/legend.tsx:
```diff
@@ -0,0 +1,203 @@
+  action: LegendAction
+): EChartsOption {
+  let finalOptions = option
+  if (change === 'roundBar' && action.type === 'toggle') {
```
Constants

ThaynanAndrey commented on packages/charts/src/components/tooltip/tooltip.tsx:
```diff
@@ -0,0 +1,83 @@
+  invert = false,
+  percentage = false,
+}: ChartTooltipProps) {
+  if (isArray(params) && invert) params.reverse()
```
```suggestion
  if (isArray(params) && invert) params.reverse()
  
```

ThaynanAndrey commented on packages/charts/src/components/tooltip/tooltip.tsx:
```diff
@@ -0,0 +1,83 @@
+  )
+}
+
+export function ChartTooltipBase({
```
Documentation

ThaynanAndrey commented on packages/charts/src/tests/purity.test.tsx:
```diff
@@ -0,0 +1,52 @@
+  )
+  unmount()
+  expect(series).toEqual([{ data: [1, 2, 3], name: 'A' }])
+  expect(xAxis).toEqual({ data: ['a', 'b', 'c'] })
```
I think it is not testing nothing 🤔

ThaynanAndrey commented on packages/charts/src/theme/chartStyles.ts:
```diff
@@ -0,0 +1,333 @@
+    //Sizing
+    right: 'center',
+    bottom: '9%',
+    height: '9%',
```
Why so specific 9%?

ThaynanAndrey commented on packages/charts/src/theme/components/tooltip.css:
```diff
@@ -0,0 +1,40 @@
+[data-sl-chart-tooltip] {
+  background-color: var(--sl-color-gray-0);
+  border-radius: 4px;
+  /* min-width: 100px; */
```
```suggestion
```

ThaynanAndrey commented on packages/charts/src/utils/hooks.ts:
```diff
@@ -0,0 +1,328 @@
+  defaultColorShade,
+} from '../theme/colors'
+
+export function applySeriesHook(
```
Documentation

leofarias-cpu committed:
fix(chart): remove commented out code

JRamonAlves committed:
fix(chart-legend): fix implementation not using the types and changes for more meaningful names

JRamonAlves committed:
fix(chart-hooks): put charts default hooks into an different archive

leofarias-cpu committed:
fix(chart-tooltip): remove unecessary important, and add white-space: normal to serie name

leofarias-cpu committed:
fix(chart-skeleton): remove unecessary default height and width parameters

theo-rodrigues committed:
style(chart): improve readability with whitespace

theo-rodrigues committed:
Merge branch 'feature/charts-package' of https://github.com/theo-rodrigues/shoreline-charts into feature/charts-package

JRamonAlves committed:
Merge branch 'feature/charts-package' of https://github.com/theo-rodrigues/shoreline-charts into feature/charts-package

JRamonAlves committed:
fix(chart-scatter): change the name of the variant to third-dimension

JRamonAlves committed:
refactor(chart-scatter): changes the name of the variant to third-dimension

JRamonAlves committed:
refactor(chart-legend): change to the shoreline button instead of normal html

JRamonAlves committed:
fix(chart): fix Flex box to 0 gap

leofarias-cpu committed:
style(chart): legend colors selection clarified

leofarias-cpu committed:
docs: update readme

Updated the README to include installation instructions, usage commands, and contribution guidelines.

leofarias-cpu committed:
style(chart): setBarGap refactored to be clearer

theo-rodrigues committed:
style(legend): change the fontsize unit from px to rem

leofarias-cpu committed:
fix(chart-compositor): bar gap now properly applied

leofarias-cpu committed:
fix(chart-compositor): bar gap added to compositor story

leofarias-cpu committed:
docs(chart): adds docs for all hooks missing them

leofarias-cpu committed:
docs(chart): adds docs for chart skeleton

leofarias-cpu committed:
docs(chart): adds docs for chart utils

leofarias-cpu committed:
feat(chart): removed buildDefaultSerie to avoid unecessary cloning

leofarias-cpu committed:
docs(chart): more docs for chart utils

leofarias-cpu committed:
docs(chart): changed name and added docs for apply series hooks

JRamonAlves committed:
fix(chart-zoom): remove zoom features from the charts

JRamonAlves committed:
fix(chart-skeleton): pass css props to css proper classes

JRamonAlves committed:
Merge branch 'feature/charts-package' of https://github.com/theo-rodrigues/shoreline-charts into feature/charts-package

JRamonAlves committed:
fix(chart-compositor): change name of function according to preview changes

leofarias-cpu committed:
fix(chart): properly merges series with defaultStyle.series

leofarias-cpu committed:
feat(chart): locale prop now has loose autocomplete of valid lcoales

JRamonAlves committed:
refactor(chart-zoom): remove the data zoom options and remove example in docs that uses it

leofarias-cpu committed:
fix(chart): removed zoom story

leofarias-cpu committed:
docs(chart-skeleton): added justification for the very exact 40px padding

leofarias-cpu committed:
style(chart): changed componenent declarations to use arrow function syntax

componenents with forward ref now use arrow function syntax to axoid repeating the name of the
component

leofarias-cpu committed:
docs(chart-tooltip): added docs and removed some duplicate code

leofarias-cpu committed:
docs: add docs for compactNumber

leofarias-cpu committed:
docs: clarified compactNumber use

leofarias-cpu committed:
style(chart): changed checkbox legend to an exhaustive switch on the action type

JRamonAlves committed:
refactor(chart-legend): change 'ifs' and 'elses' to a switch statement to guarantee type coverage

leofarias-cpu committed:
docs(chart-legend): added docs for legendItem

leofarias-cpu committed:
docs(chart-tooltip): added docs for tooltipBase

leofarias-cpu committed:
style(chart-tooltip): chartToolTipBase -> chartTooltipLine

leofarias-cpu committed:
style(chart-legend): renamed legend action to be simpler and added docs to checkAllSelected

theo-rodrigues committed:
refactor(legend): refactor the legend changes handler

theo-rodrigues committed:
fix(chart): remove getchanges and add the proper way to handle the legend changes

theo-rodrigues committed:
refactor(legend): improve legend hooks design

theo-rodrigues committed:
fix(chart): change handlechange to handleHooks

leofarias-cpu committed:
fix(chart-legend): barRoundingToggle now handles cases with 0 visible series

leofarias-cpu committed:
style(chart-legend): removed 1 level of identation and added some comments

theo-rodrigues committed:
docs(legend): add legend remaining docs

leofarias-cpu committed:
docs(chart-legend): add doc for toggleSerie

leofarias-cpu committed:
fix(chart-legend): legend button checkbox now properly centered in the docs

lucasaarcoverde commented on packages/charts/src/theme/components/chart.css:
```diff
@@ -0,0 +1,11 @@
+[data-sl-chart] {
```
can you check the css files and change from `px` to `rem`?

leofarias-cpu committed:
refactor(chart): replaces lodash dependency with lodas.clonedeep, the only method we actually need

leofarias-cpu committed:
docs(chart): add default to locale prop

leofarias-cpu committed:
Merge branch 'main' of https://github.com/theo-rodrigues/shoreline-charts
